flowetl/tests/integration/conftest.py

```diff
@@ -0,0 +1,168 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+# -*- coding: utf-8 -*-
+"""
+Conftest for flowetl integration tests
+"""
+import os
+import shutil
+
+from time import sleep
+from subprocess import DEVNULL, Popen
+from pendulum import now, Interval
+from airflow.models import DagRun
+
+import pytest
+import docker
+
+
+@pytest.fixture(scope="session")
+def docker_client():
+    """
+    docker client object
+    """
+    return docker.from_env()
+
+
+@pytest.fixture(scope="session")
+def flowetl_tag():
+    """
+    Get flowetl tag to use
+    """
+    return os.environ.get("FLOWETL_TAG", "latest")
+
+
+# pylint: disable=redefined-outer-name
+@pytest.fixture(scope="module")
+def flowetl_container(docker_client, flowetl_tag):
+    """
+    Fixture that starts a running flowetl container and
+    yeilds the container object.
+    """
+    container = docker_client.containers.run(
+        f"flowminder/flowetl:{flowetl_tag}", detach=True
+    )
+    # breif sleep to wait for backing DB to be ready
+    sleep(2)
```

Comment:
I would really like to avoid random sleeps as much as possible if we can. Almost every instance of hard-coded timeouts I have seen in practice is either flaky because it's on the verge of being too short, or rather long/conservative and thus wastes time during tests or in manual usage.

Given that in general we want reliable ways of verifying that our containers have started up successfully anyway (see my comments in #745), it would be great if the tests and fixtures could also use these.

I'm ok to defer this to a follow-up PR if it's too tedious now, but would like to tackle this asap.

Reply:
I feel you! But not clear how to know when it's ready...if you have a suggestion?

Reply:
Parse the logs? That feels hacky? Of course could completely build our own Airflow container and ignore puckel et al...

Reply:
Yeah, let's think about it properly and worry about this in a separate PR.

Reply:
https://github.com/Flowminder/FlowKit/issues/750